compiler/rustc_middle/src/ty/instance.rs

```diff
@@ -332,6 +334,26 @@ impl<'tcx> InstanceDef<'tcx> {
                     .map_or_else(|| adt_def.is_enum(), |dtor| tcx.cross_crate_inlinable(dtor.did))
             });
         }
+        if let ty::InstanceDef::AsyncDropGlueCtorShim(.., Some(ty)) = *self {
```

Comment:
Deduplicate with the above sync drop check by checking the instance def again to choose whether to call `destructor` or `async_destructor`

Reply:
Done in 1a8eae1aba998af1b0fdc597c0e263d4d391b169